app/components/gh-file-uploader.js

```diff
@@ -150,10 +170,22 @@ export default Component.extend({
 
     actions: {
         fileSelected(fileList) {
-            this.set('file', fileList[0]);
-            run.schedule('actions', this, function () {
-                this.generateRequest();
-            });
+            // can't use array destructuring here as FileList is not a strict
+            // array and fails in Safari
+            // jscs:disable requireArrayDestructuring
+            let file = fileList[0];
+            // jscs:enable requireArrayDestructuring
+            let validationResult = invokeAction(this, 'validate', file);
```

Comment:
That would fail if the we have a validate action and it returns `false` for something which isn't what the default validation checks, it may be possible if we use a long-form conditional and explicitly check for `undefined` but then we could probably just check for the existence of `this.get('validate')`  🤔 - I'll try that route